chain/client/src/sync/epoch.rs

```diff
@@ -464,12 +521,28 @@ impl EpochSync {
                 tracing::warn!("Ignoring epoch sync proof from unexpected peer: {}", source_peer);
                 return Ok(());
             }
-            if proof.current_epoch.first_block_header_in_epoch.height()
-                + self.config.epoch_sync_accept_proof_max_horizon
+            if proof
+                .current_epoch
+                .first_block_header_in_epoch
+                .height()
+                .saturating_add(chain.epoch_length.min(chain.transaction_validity_period))
```

Comment:
Why not max? From the description I got that we don't want to sync epoch from peer which is `< transaction_validity_period` ahead.

Reply:
Good catch! Unfortunately this isn't tested. I'm not sure how to exactly test this because it requires the responding node to give it a bad proof. I guess when testing the proof verification code I'd have to figure that out.